### PR TITLE
JXplorer: init at 3.3.1.2

### PR DESCRIPTION
**Pull request by benwbooth**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/jxplorer/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    install -Dm755 "${pname}" "$out/bin/${pname}"
+    cp -r "${jxplorerItem}/." "$out"
+    install -Dm644 images/JX128.png "$out/share/pixmaps/${pname}.png"
+  '';
+
+  meta = with lib; {
+    description = "A Java Ldap Browser";
+    homepage    = "https://sourceforge.net/projects/jxplorer/";
+    license     = "CA Open Source Licence Version 1.0";
+    maintainers = with maintainers; [ benwbooth ];
```
Comment (cole-h) [inline pkgs/applications/networking/jxplorer/default.nix:43]: FYI: you need to add yourself as a maintainer in the maintainers list:

https://github.com/NixOS/nixpkgs/blob/62d571148a24889811b8f4cdb69c05eb00d260ba/maintainers/maintainer-list.nix#L1

**Review by cole-h — CHANGES_REQUESTED** [review]

**Comment by benwbooth** [comment]
I did add myself as a maintainer in a previous pull request: https://github.com/NixOS/nixpkgs/pull/209776.

Should I add myself again to this pull request or create a separate pull request to add myself as a maintainer?

**Comment by cole-h** [comment]
Considering that PR was never merged, you should add it in this PR :P 

It does need to be a separate commit, though. `maintainers: add benwbooth` should be the commit message.

**Comment by benwbooth** [comment]
Done. Thanks!

File: pkgs/applications/networking/jxplorer/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    cp -r ./. "$out/opt/jxplorer"
+    install -Dm755 "${pname}" "$out/bin/${pname}"
+    cp -r "${jxplorerItem}/." "$out"
+    install -Dm644 images/JX128.png "$out/share/pixmaps/${pname}.png"
+  '';
+
+  meta = with lib; {
+    description = "A Java Ldap Browser";
+    homepage    = "https://sourceforge.net/projects/jxplorer/";
+    license     = "CA Open Source Licence Version 1.0";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/jxplorer/default.nix:42]: Should we add this to lib.licenses?

File: pkgs/applications/networking/jxplorer/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    icon = "jxplorer";
+  };
+
+  configurePhase = ''
+    cat >"${pname}" << EOF
+    #!/bin/sh
+    cd "$out/opt/jxplorer"
+    export JAVA_HOME="${jdk8}"
+    sh jxplorer.sh "\$@"
+    EOF
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/jxplorer/default.nix:27]: Please use makeWrapper

File: pkgs/applications/networking/jxplorer/default.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    chmod +x "${pname}"
+  '';
+
+  installPhase = ''
+    install -d "$out/opt/jxplorer" "$out/bin" "$out/share/pixmaps" "$out/share/applications"
+    cp -r ./. "$out/opt/jxplorer"
+    install -Dm755 "${pname}" "$out/bin/${pname}"
+    cp -r "${jxplorerItem}/." "$out"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/jxplorer/default.nix:35]: Please use copyDesktopItems

File: pkgs/applications/networking/jxplorer/default.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{ lib, stdenv, fetchurl, makeDesktopItem, ant, jdk8 }:
+
+stdenv.mkDerivation rec {
+  pname   = "jxplorer";
+  version = "3.3.1.2";
+
+  src = fetchurl {
+    url    = "https://github.com/pegacat/${pname}/releases/download/v${version}/${pname}-${version}-project.tar.bz2";
+    sha256 = "/lWkavH51OqNFSLpgT+4WcQcfW3WvnnOkB03jB7bE/s=";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/jxplorer/default.nix:9]:
```suggestion
    hash = "sha256-/lWkavH51OqNFSLpgT+4WcQcfW3WvnnOkB03jB7bE/s=";
```

File: pkgs/applications/networking/jxplorer/default.nix [hunk 6]
```diff
@@ -0,0 +1,46 @@
+{ lib, stdenv, fetchurl, makeDesktopItem, ant, jdk8 }:
+
+stdenv.mkDerivation rec {
+  pname   = "jxplorer";
+  version = "3.3.1.2";
+
+  src = fetchurl {
+    url    = "https://github.com/pegacat/${pname}/releases/download/v${version}/${pname}-${version}-project.tar.bz2";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/jxplorer/default.nix:8]:
```suggestion
    url    = "https://github.com/pegacat/jxplorer/releases/download/v${version}/jxplorer-${version}-project.tar.bz2";
```

File: pkgs/applications/networking/jxplorer/default.nix (hunk 2 above)
Comment (benwbooth) [reply pkgs/applications/networking/jxplorer/default.nix:45]: I couldn't find a url or spdxId for this license. I can't find any other software that uses it. Is it OK to add it lib.licenses without a url or spdxId?

File: pkgs/applications/networking/jxplorer/default.nix (hunk 1 above)
Comment (benwbooth) [reply pkgs/applications/networking/jxplorer/default.nix:46]: already merged from a previous pull request

File: pkgs/applications/networking/jxplorer/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/networking/jxplorer/default.nix:46]: license is available at http://jxplorer.org/licence.html and https://www.openhub.net/licenses/caossl, I think we can shorten it with caossl

File: pkgs/applications/networking/jxplorer/default.nix [hunk 7]
```diff
@@ -0,0 +1,50 @@
+{ lib, stdenv, fetchurl, makeDesktopItem, ant, jdk8, copyDesktopItems, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  pname   = "jxplorer";
+  version = "3.3.1.2";
+
+  src = fetchurl {
+    url    = "https://github.com/pegacat/jxplorer/releases/download/v${version}/jxplorer-${version}-project.tar.bz2";
+    hash = "sha256-/lWkavH51OqNFSLpgT+4WcQcfW3WvnnOkB03jB7bE/s=";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/jxplorer/default.nix:9]:
```suggestion
  pname = "jxplorer";
  version = "3.3.1.2";

  src = fetchurl {
    url = "https://github.com/pegacat/jxplorer/releases/download/v${version}/jxplorer-${version}-project.tar.bz2";
    hash = "sha256-/lWkavH51OqNFSLpgT+4WcQcfW3WvnnOkB03jB7bE/s=";
```

**Comment by SuperSandro2000** [comment]
@ofborg build jxplorer

**Pull request merged**